### DISCUSSION TITLE
how to update image macos version ?

### DISCUSSION THREAD

**Discussion by alexxxcs1** [cirruslabs/tart#631] [Q&A] [answered]
i create a local vm image from ipsw long long ago and install many application on this image,
now i want to upgrade macos version to Sonoma or other higher version, but i got this 'Faild to personalize the software update, please try again'

how to update image macos version ? it's possible ?

![image](https://github.com/alexxxcs1/resource/blob/master/0zm5w12000cekx3z55498.png?raw=true)

**Comment by fkorotkov** [comment, marked as answer]
Have you used [Tart Packer plugin](https://github.com/cirruslabs/packer-plugin-tart) on the image? If so the VM is missing recovery partition because the plugin removes it to save space which lead to this error.

If you created the VM from an IPSW then the partition should be there. I've seen people successuflly installing updates via command like:

```bash
sudo softwareupdate --list
sudo softwareupdate -ir "macOS Ventura 13.6-22G120"
```

Also in case you'll try the above command you'll see a more detailed message in case of an error.

**Reply by alexxxcs1** [reply]
@fkorotkov 
thanks for reply! 
yes,i create vm from "tart create [name] --from-ipsw=[ipswpath]" and use packer to install other application.
so did i use recovery mode DiskUtil to create a recovery partition can fix it?

**Reply by fkorotkov** [reply]
Unfortunately it seems the recovery partition is not recoverable now. At least we don't know an easy way to do it.

Since you used Packer to automate installation of the application is it possible you can recreate the VM? Or you have local state that you want to preserve?

**Reply by alexxxcs1** [reply]
yes , i want to preserve all local state, i just want to upgrade to snoma and use xcode 15.x.
recreate vm have too many work todo , Even though I kept the pkr.hcl configuration, there are still a lot of small hardcode config.
and the network environment is an intranet, so download application will take long time.
so there has any way to upgrade macos version in this situation? or other keep local state way ?

**Reply by fkorotkov** [reply]
Unfortunately we are not aware of a way to restore the partition.  Sorry for the bad news. Hopefully someone else might know of a workaround.

**Reply by alexxxcs1** [reply]
ok , thank u very mach , i will try recreate a new image without packer plugin.
